connectors/sources/sharepoint.py

```diff
@@ -0,0 +1,715 @@
+            site_url (str): Site path of sharepoint
+            timestamp (timestamp, optional): Timestamp of item last modified. Defaults to None.
+            doit (boolean, optional): Boolean value for whether to get content or not. Defaults to None.
+
+        Returns:
+            dictionary: Content document with id, timestamp & text.
+        """
+
+        if not (self.enable_content_extraction and doit and document["size"]):
+            return
+
+        document_size = int(document["size"])
+        if document_size > FILE_SIZE_LIMIT:
+            logger.warning(
+                f"File size {document_size} of file {document['title']} is larger than {FILE_SIZE_LIMIT} bytes. Discarding file content"
+            )
+            return
+
+        source_file_name = ""
+
+        async with NamedTemporaryFile(mode="wb", delete=False) as async_buffer:
+            async for response in self._api_call(
+                url_name=ATTACHMENT,
+                host_url=self.host_url,
+                value=site_url,
+                file_relative_url=file_relative_url,
+            ):
+                async for data in response.content.iter_chunked(CHUNK_SIZE):
+                    await async_buffer.write(data)
+            source_file_name = async_buffer.name
+
+        await asyncio.to_thread(
+            convert_to_b64,
+            source=source_file_name,
+        )
+        async with aiofiles.open(file=source_file_name, mode="r") as target_file:
+            # base64 on macOS will add a EOL, so we strip() here
+            attachment_content = (await target_file.read()).strip()
+        await remove(source_file_name)
+        return {
+            "_id": document.get("id"),
+            "_timestamp": document.get("_timestamp"),
+            "_attachment": attachment_content,
+        }
+
+    async def invoke_get_call(self, site_url, param_name, list_id=None):
+        """Invokes a GET call to the Sharepoint Server/Online.
+
+        Args:
+            site_url(string): site url to the sharepoint farm.
+            param_name(string): parameter name whether it is SITES, LISTS, DRIVE_ITEM, LIST_ITEM.
+            list_id(string, optional): Id of list item or drive item. Defaults to None.
+        Yields:
+            Response of the GET call.
+        """
+        skip, top = 0, 5000
+        next_url = ""
+        while True:
+            if param_name in [SITES, LISTS]:
+                response = await anext(
+                    self._api_call(
+                        url_name=param_name,
+                        parent_site_url=site_url,
+                        skip=skip,
+                        top=top,
+                        host_url=self.host_url,
+                    )
+                )
+                response_result = response.get("value", [])
+                yield response_result
+
+                skip += 5000
+                if len(response_result) < 5000:
```

Comment:
Do they guarantee to always return 5000 documents if there are documents left? Just asking cause for Salesforce their page sizes are dynamic depending on size of payload, so wanted to double check here too.

Reply:
yes it is mentioned here:  https://alextofan.com/2019/08/22/how-to-get-more-than-5000-item-from-sharepoint-online-in-flow/